ros2_node/lh_tracker/lh_tracker_geometry/lh_tracker_geometry_node.py

```diff
@@ -0,0 +1,832 @@
+#!/usr/bin/env python3
+
+import rclpy
+import numpy as np
+import cv2
+import random
+import copy
+import lh_tracker_geometry.constants as csts
+
+from tf2_kdl import PyKDL, do_transform_frame, transform_to_kdl, do_transform_vector
+from rclpy.node import Node
+from lh_tracker_msgs.msg import LHtracker
+from geometry_msgs.msg import Point, TransformStamped
+from visualization_msgs.msg import Marker, MarkerArray
+from std_msgs.msg import ColorRGBA
+from lh_tracker_geometry.lh_geometry import LH2Geometry
+from tf2_ros import TransformBroadcaster, StaticTransformBroadcaster
+
+from scipy.optimize import least_squares
+from datetime import datetime
+
+
+class LH_tracker_geometry(Node):
+    def __init__(self):
+        """Init LH_tracker_geometry node"""
+        super().__init__("lh_tracker_geometry")
+        self.subscription = self.create_subscription(
+            LHtracker, "lh_msgs", self.lh_sub_callback, 10
+        )
+
+        self.possible_pairs = [[0, 1], [2, 3], [4, 5], [6, 7]]
+        self.geometry = LH2Geometry()
+
+        self.marker_estimated_pose = Marker()
+        self.marker_estimated_pose.id = 1
+        self.marker_estimated_pose.header.frame_id = "map"
+        self.marker_estimated_pose.type = 0  # arrow
+        self.marker_estimated_pose.scale.x = 0.3
+        self.marker_estimated_pose.scale.y = 0.1
+        self.marker_estimated_pose.scale.z = 0.1
+        self.marker_estimated_pose.color.a = 1.0
+        self.marker_estimated_pose.lifetime.sec = 5
+        self.marker_estimated_pose.color.r = 1.0
+        self.init_line_list()
+        self.sensor_height = 0.415
+        self.init_sensors_markers()
+        self.lh_lines_pb = self.create_publisher(Marker, "lh_lines", 10)
+        self.lh_est_pose_pb = self.create_publisher(Marker, "lh_est_pose", 10)
+        self.sensors_est_position_pb = self.create_publisher(
+            Marker, "sensors_est_position", 10
+        )
+        self.tf_broadcaster = TransformBroadcaster(self)
+
+        self.avg_nb = 200
+        self.global_index = 0
+        self.calibration_done = False
+
+        self.initial_tf = self.kdl_to_transform(
+            PyKDL.Frame(
+                R=PyKDL.Rotation().EulerZYX(np.deg2rad(45), 0.0, 0.0),
+                V=PyKDL.Vector(x=1.5, y=0.1, z=self.sensor_height),
+            )
+        )
+
+        self.saved_iteration_list = LHtracker()
+
+        # self.create_timer(959 / 48000, self.loop)
```

Comment:
Remove commentaries of this file.

Reply:
[Done.](https://github.com/robotique-ecam/ViveTracker/pull/18/commits/ae344ba3c0c1352b1690df0c98f572dbfb3009cc)